### ISSUE TITLE
Projected volume does not support configMap

### ISSUE THREAD

**Issue by ivanjaros** [kubernetes/kubernetes#46965]
According to the documentation in here https://kubernetes.io/docs/concepts/storage/volumes/#projected the configMap should be supported in the projected volume, yet only secrets work. 

Using Kubernetes 1.6.4 on GKE.

(I have a bunch of configmaps that I want to mount into a single directory so this is very important for me.)

**Comment by xiangpengzhao** [comment]
@kubernetes/sig-storage-misc

**Comment by xingzhou** [comment]
I'd like to take a look at this issue, @ivanjaros, mind paste more details on the reproduce steps?

**Comment by ivanjaros** [comment]
I have defined the projected volume with one configmap in it and I have ended up with a volume of type emptyDir. Then I manually edited the manifest though the admin and set the config map but I got error so I tried secret and the finally worked.

Does not work:
```
volumes:
  - name: test
    projected:
      sources:
      - configMap:
          name: myconfigmap
```

works
```
volumes:
  - name: test
    projected:
      sources:
      - secret:
          name: mysecret
```

**Comment by xingzhou** [comment]
@ivanjaros, I tried the following steps on v1.6.4 branch and master branch, and all worked well:
1. Create a config map like:
```
apiVersion: v1
kind: ConfigMap
metadata:
  name: dongdong
  namespace: default
data:
  special.level: very
  special.type: charm
```

2. Create a pod using the following yam:
```
apiVersion: v1
kind: Pod
metadata:
  name: volume-test
spec:
  containers:
  - name: container-test
    imagePullPolicy: Never
    image: nginx
    volumeMounts:
    - name: all-in-one
      mountPath: "/projected-volume"
      readOnly: true
  volumes:
  - name: all-in-one
    projected:
      sources:
      - configMap:
          name: dongdong
          items:
            - key: special.level
              path: config
```

3. Login to the pod and check the `/projected-volume` folder, I can see the config file there with content `very`

Are the above steps the ones you are using in your env? or can you show the error messages you have?

**Comment by ivanjaros** [comment]
Hm, my first thought was the absence of items for configmap in my case but I have tried that as well yesterday and no success. But since it works for you it means there is something else wrong in my case and Kubernetes is fine. So I have to figure out what is causing this.

Thanks for your time.

**Comment by ivanjaros** [comment]
I have to re-open this. I spend another half a day trying to figure this out but still no success. ConfigMap volume just does not work on GKE. I tried multiple clusters, different machines, I tried to enable alpha features but still end up with emptyDir.

Also I found some references like #26223 or #23392 but those are quire old and marked as closed.

**Comment by liggitt** [comment]
@ivanjaros what version of kubectl are you using? (`kubectl version`)

**Comment by ivanjaros** [comment]
1.5.3(gcloud, win) but I used UI to create the deployment and edited the deployment manually through ui as well.

**Comment by jpeeler** [comment]
The projected volume driver wasn't released until 1.6. It's unclear to me if you upgraded all of Kubernetes or not, but if it was just kubectl projected volumes won't work.

**Comment by ivanjaros** [comment]
I haven't worked with k8s for a while so dunno if it is fixed or not but since it seems a minor issue and maybe only local

/close
